lib/common/xml_display.c

```diff
@@ -253,34 +253,32 @@ pcmk__xml_show(pcmk__output_t *out, int log_level, const char *prefix,
 
 /*!
  * \internal
- * \brief Log XML portions that have been marked as changed
+ * \brief Output XML portions that have been marked as changed
  *
- * \param[in] log_level  Priority at which to log the messages
- * \param[in] data       XML node to log
- * \param[in] depth      Current indentation level
- * \param[in] options    Group of \p pcmk__xml_fmt_options flags
+ * \param[in,out] out      Output object
+ * \param[in]     data     XML node to output
+ * \param[in]     depth    Current indentation level
+ * \param[in]     options  Group of \p pcmk__xml_fmt_options flags
  *
- * \note This is a recursive helper for \p pcmk__xml_log_changes(), logging
+ * \note This is a recursive helper for \p pcmk__xml_log_changes(), showing
  *       changes to \p data and its children.
  */
 static void
-log_xml_changes_recursive(int log_level, const xmlNode *data, int depth,
-                          uint32_t options)
+show_xml_changes_recursive(pcmk__output_t *out, const xmlNode *data, int depth,
+                           uint32_t options)
 {
     /* @COMPAT: When log_data_element() is removed, we can remove the options
      * argument here and instead hard-code pcmk__xml_log_pretty.
      */
-    xml_node_private_t *nodepriv = NULL;
+    xml_node_private_t *nodepriv = (xml_node_private_t *) data->_private;
```

Comment:
Whoops, moved it before the NULL check... that's why it wasn't up there already. I'll fix that.

Reply:
Actually, the caller already asserts non-NULL. I can just scrap the NULL check.